### PR TITLE
Disallow compiling the C extension when running with PyPy

### PR DESCRIPTION
**Pull request by thedrow**
Also added missing PyPi trove classifiers for CPython and PyPy.

### CODE REVIEW

**Comment by thedrow** [comment]
Sorry about the failure. I can rebase if needed.

**Comment by emmett9001** [comment]
@thedrow Looks like the PyPy failures are caused by attempting to import rdkafka-related test files that shouldn't even be imported. An easy way to fix this might be to wrap the [`_rd_kafka` imports](https://github.com/Parsely/pykafka/blob/2.2.0/pykafka/rdkafka/producer.py#L8) in `try..except`. It would probably be better to not run the tests under PyPy altogether, but I'm not sure how easy Travis will make that for us.

File: setup.py [hunk 1]
```diff
@@ -24,6 +25,11 @@
 from setuptools.command.test import test as TestCommand
 from setuptools.extension import Extension
 
+if sys.version_info < (2, 7):
+    raise Exception('pykafka requires Python 2.7 or higher.')
+
+python_implementation = platform.python_implementation()
+cpython = python_implementation == 'CPython'
```
Comment (emmett9001) [inline setup.py]: A better variable name here would be `is_cpython`

**Comment by emmett9001** [comment]
Aside from my comment and the failing tests, this looks great. Let me know if you manage to make the PyPy tests pass. Thanks!

**Comment by thedrow** [comment]
So the tests hang again. See https://travis-ci.org/Parsely/pykafka/jobs/107402230#L496

**Comment by thedrow** [comment]
@emmett9001 Can you check the tests again? This has nothing to do with PyPy.

File: setup.py [hunk 2]
```diff
@@ -162,6 +174,8 @@ def run_setup(with_rdkafka=True):
     )
 
 try:
+    if not is_cpython:
+        raise ve_build_ext.UnsupportedPlatform(python_implementation)
```
Comment (emmett9001) [inline setup.py]: Would this `raise..except` pair make more sense as a simple boolean check on `is_cpython`? It seems like that would make the logic clearer.

File: tests/pykafka/rdkafka/test_producer.py [hunk 3]
```diff
@@ -3,11 +3,16 @@
...
+try:
+    from pykafka.rdkafka import _rd_kafka
+    RDKAFKA = True
+except ImportError:
+    RDKAFKA = False # C extension not built
 
 @pytest.mark.skipif(platform.python_implementation() == "PyPy",
                     reason="We pass PyObject pointers as msg_opaques for "
                            "delivery callbacks, which is unsafe on PyPy.")
+@pytest.mark.skipif(not RDKAFKA, reason="C extension for librdkafka not built.")
```
Comment (emmett9001) [inline tests/pykafka/rdkafka/test_producer.py]: Remove trailing whitespace

File: tests/pykafka/rdkafka/test_rd_kafka_consumer.py [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
 from pykafka.exceptions import RdKafkaStoppedException, RdKafkaException
-from pykafka.rdkafka import _rd_kafka
+try:
+    from pykafka.rdkafka import _rd_kafka
+    RDKAFKA = True
+except ImportError:
+    RDKAFKA = False # C extension not built
 from pykafka.test.utils import get_cluster, stop_cluster
 from pykafka.utils.compat import get_bytes
+import pytest
```
Comment (emmett9001) [inline tests/pykafka/rdkafka/test_rd_kafka_consumer.py]: This import should be above, in the same group as `unittest2`

File: tests/pykafka/rdkafka/test_rd_kafka_consumer.py [hunk 5]
```diff
@@ -1,12 +1,20 @@
...
 
-
+@pytest.mark.skipif(platform.python_implementation() == "PyPy",
+                    reason="librdkafka is not supported with PyPy.")
```
Comment (emmett9001) [inline tests/pykafka/rdkafka/test_rd_kafka_consumer.py]: I actually don't think we need to skip this test under PyPy - it's passing in master.

**Comment by emmett9001** [comment]
@thedrow Yup, the tests are looking better now. I left a few minor comments.

**Comment by emmett9001** [comment]
@thedrow After the comments I've left are addressed, this will be mergeable.

File: tests/pykafka/rdkafka/test_rd_kafka_consumer.py (hunk 4 above)
Comment (thedrow) [inline tests/pykafka/rdkafka/test_rd_kafka_consumer.py]: isort says otherwise. I can change that though.

**Comment by emmett9001** [comment]
Looks like this pull request is still using the naive method for test inheritance. See the docstring [here](https://github.com/Parsely/pykafka/blob/bb8363392f665b4e322f5e7bb7e540df31bea3cd/tests/pykafka/__init__.py#L4) for the proper way to do test inheritance, as well as an example [here](https://github.com/Parsely/pykafka/blob/bb8363392f665b4e322f5e7bb7e540df31bea3cd/tests/pykafka/test_balancedconsumer.py#L482) and [here](https://github.com/Parsely/pykafka/blob/bb8363392f665b4e322f5e7bb7e540df31bea3cd/tests/pykafka/test_balancedconsumer.py#L385).

**Comment by emmett9001** [comment]
@thedrow If you'd like this pull request to get merged, please address the above comment about test inheritance. After that's done, this should be mergeable, but it won't be until then.

**Comment by emmett9001** [comment]
Closed in favor of #715

**Pull request closed without merging**